doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,289 @@
+
+                issue = open_issues.pop()
+
+                if not self.dry_run:
+                    # Keep notify as False since this description will constantly be updated everytime there's a
+                    # new build
+                    issue.update(fields=fields, notify=False)
+                    self.runtime.logger.info(f"The fields of {issue.key} has been updated to {fields}")
+                else:
+                    self.runtime.logger.info(f"[DRY RUN]: Would have updated {issue.key} with to fields: {fields}")
+
+            else:
+                self.runtime.logger.error(f"More than one JIRA ticket exists: {open_issues}")
+
+    def get_packages_with_vulnerabilities(self, mapping: dict):
+        """
+        Check the latest scan of a particular package for a version (which we already found) and check if
+        vulnerabilities exist. Collect those and return back.
+
+        :param mapping: Eg: {'ose-network-tools-container': {'package_name_with_version': 'ose-network-tools-container-v4.15.0-',
+                                                           'nvr': 'ose-network-tools-container-v4.15.0-202310170244.p0.gdf85e45.assembly.stream',
+                                                           'package_name': 'ose-network-tools-container',
+                                                           'latest_coverity_scan': '337873'}
+        }
+        """
+
+        nvrs_with_vulnerabilities = {}
+
+        # Check if vulnerabilities detected for a NVR
+        for brew_package_name, data in mapping.items():
+            if not data["latest_coverity_scan"]:
+                self.runtime.logger.info(f"No successful scan found for package {brew_package_name}")
+                continue
+
+            if self.check_if_vulnerabilities_exist(task_id=data["latest_coverity_scan"], nvr=data["nvr"]):
+                nvrs_with_vulnerabilities[brew_package_name] = data
+
+            else:
+                self.runtime.logger.info(f"No vulnerabilities found for NVR: {data['nvr']}")
+
+        return nvrs_with_vulnerabilities
+
+    async def ocp_bugs_workflow_run(self, nvrs: list):
+        # List of brew packages to check
+        brew_package_names = []
+
+        for nvr in nvrs:
+            brew_info = self.koji_session.getBuild(nvr)
+
+            # Get ose-network-tools-container-v4.15.0 from ose-network-tools-container-v4.15.0-202310170244.p0.gdf85e45.assembly.stream
+            brew_package_names.append({
+                "package_name_with_version": brew_info["nvr"].split(f"v{self.version}")[0] + f"v{self.version}.0",
+                "nvr": brew_info["nvr"],
+                "brew_build_id": brew_info["id"],
+                "package_name": brew_info["package_name"]
+            })
+
+        # Get latest scan results
+        brew_coverity_mapping = {}
+
+        for package in brew_package_names:
+            package["latest_coverity_scan"], package["previous_scan_id"], package["previous_scan_nvr"] = \
```

Comment:
I think we can make this simpler by passing package to get_scan_info and then it fills the fields in

Reply:
Hmm, I was hoping to keep it like this for now. Want to refine the entire workflow anyways. Shall I keep this as a refinement?

Reply:
It's more of a comment about code style.. I don't know if creating a jira story is worth it

Reply:
Yeah I'm thinking about keep it like this for now. There could be some changes coming in, like openscanhub is being migrated to DefectDojo soon, so will have to change stuff anyways.

Reply:
Makes sense